core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -135,7 +136,12 @@ private[spark] trait Logging {
         val replLevel = Option(replLogger.getLevel()).getOrElse(Level.WARN)
         if (replLevel != rootLogger.getEffectiveLevel()) {
           System.err.printf("Setting default log level to \"%s\".\n", replLevel)
-          System.err.println("To adjust logging level use sc.setLogLevel(newLevel).")
+          if (SparkSubmit.isRShell) {
```

Comment:
+1 -- I personally find it odd that we would import `deploy.SparkSubmit` into `Logging` which is a base class used throughout the code base. We could also change the log message to be more vague. Something like `To adjust logging level please call setLogLevel(newLevel) using SparkContext`  might work for all languages ?